src/androidTest/java/database/GeneralInfoDaoTest.java

```diff
@@ -0,0 +1,102 @@
+package database;
+
+import android.content.Context;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.split.android.client.storage.db.GeneralInfoEntity;
+import io.split.android.client.storage.db.SplitRoomDatabase;
+
+public class GeneralInfoDaoTest {
+
+    SplitRoomDatabase mRoomDb;
+    Context mContext;
+
+    @Before
+    public void setup() {
+        mContext = InstrumentationRegistry.getInstrumentation().getContext();
+        mRoomDb = SplitRoomDatabase.getDatabase(mContext, "encripted_api_key");
+        mRoomDb.clearAllTables();
+    }
+
+    @Test
+    public void insertRetrieve() throws InterruptedException {
```

Comment:
we are missing checks when you try to insert `null` info, key is empty, key is null, key is has blank spaces. if some of theme trigger an exception, can we check the expected exception?

Reply:
Hey @mmelograno thanks for the review. You really have point.Passing null values to any DAO method will produce a NPE or some SQLiteException related exception depending on if the value is for a full object or a NonNull field. Both are non checked exceptions and we can handle them if necessary.
But I think the best way to handle this would be to add defensive code on any method using this classes.

Reply:
So, let's proceed by adding those (null, etc) test cases in the part of the code that we will use these variables and check if the defensive code is working properly.